test/parallel/sdk.API.claimRewards.test.ts

```diff
@@ -0,0 +1,450 @@
+  });
+
+  const rewardsLiqId1After = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[0]
+  );
+
+  const rewardsLiqId2After = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[12]
+  );
+
+  expect(rewardsLiqId1Before.rewardsAlreadyClaimed).bnEqual(BN_ZERO);
+  expect(rewardsLiqId1After.rewardsAlreadyClaimed).bnGt(BN_ZERO);
+  expect(rewardsLiqId2Before.rewardsAlreadyClaimed).bnEqual(BN_ZERO);
+  expect(rewardsLiqId2After.rewardsAlreadyClaimed).bnGt(BN_ZERO);
+
+  await promotePool(sudo.keyRingPair, liqIds[12], 20);
+});
+
+test("GIVEN an user has available some rewards in two “pools” ( one solo token, one pool ) WHEN claims all rewards THEN the user gets the rewards for that's pools", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(poolTokenIds[0], testUser1, defaultCurrencyValue),
+    Assets.mintToken(soloTokenId, testUser1, defaultCurrencyValue),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.mintLiquidity(
+        MGA_ASSET_ID,
+        poolTokenIds[0],
+        defaultCurrencyValue.divn(2)
+      )
+    )
+  );
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.promotePool(soloTokenId.toNumber(), 20),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.activateLiquidity(soloTokenId, defaultCurrencyValue.divn(2))
+    )
+  );
+
+  await waitForRewards(testUser1, soloTokenId);
+
+  const rewardsLiqId1Before = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[0]
+  );
+
+  const rewardsLiqId2Before = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    soloTokenId
+  );
+
+  await claimRewardsAll(testUser1).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+
+  const rewardsLiqId1After = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[0]
+  );
+
+  const rewardsLiqId2After = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    soloTokenId
+  );
+
+  expect(rewardsLiqId1Before.rewardsAlreadyClaimed).bnEqual(BN_ZERO);
+  expect(rewardsLiqId1After.rewardsAlreadyClaimed).bnGt(BN_ZERO);
+  expect(rewardsLiqId2Before.rewardsAlreadyClaimed).bnEqual(BN_ZERO);
+  expect(rewardsLiqId2After.rewardsAlreadyClaimed).bnGt(BN_ZERO);
+});
+
+test("GIVEN a user that has available some rewards in ten pools (max for automatically claiming) WHEN claims all rewards THEN the user gets the rewards for that's pools", async () => {
+  await createMultiplePoolsForUser(testUser1, 10);
+
+  await waitForRewards(testUser1, liqIds[9]);
+
+  const rewardsLiqIdBefore = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIds[9]
+  );
+
+  await claimRewardsAll(testUser1).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+
+  const rewardsLiqIdAfter = await getRewardsInfo(
```

Comment:
I think we shoudl check all of them, right? :thinking:

Reply:
I decided that we can only check on the first and/or last pool, but of course, I can do it for all of them

Reply:
I think that doing it in a loop is not so costly, so plz add them :)

Reply:
Added